src/interpreter/fvm2.rs

```diff
@@ -31,34 +33,43 @@ use crate::interpreter::resolve_to_key_addr;
 
 pub struct ForestExternsV2<DB> {
     rand: Box<dyn Rand>,
+    heaviest_tipset: Arc<Tipset>,
     epoch: ChainEpoch,
     root: Cid,
-    lookback: Box<dyn Fn(ChainEpoch) -> anyhow::Result<Cid>>,
-    db: DB,
+    chain_store: Arc<ChainStore<DB>>,
     chain_config: Arc<ChainConfig>,
     bail: AtomicBool,
 }
 
-impl<DB: Blockstore> ForestExternsV2<DB> {
+impl<DB: Blockstore + Send + Sync + 'static> ForestExternsV2<DB> {
     pub fn new(
         rand: impl Rand + 'static,
+        heaviest_tipset: Arc<Tipset>,
         epoch: ChainEpoch,
         root: Cid,
-        lookback: Box<dyn Fn(ChainEpoch) -> anyhow::Result<Cid>>,
-        db: DB,
+        chain_store: Arc<ChainStore<DB>>,
         chain_config: Arc<ChainConfig>,
     ) -> Self {
         ForestExternsV2 {
             rand: Box::new(rand),
+            heaviest_tipset,
             epoch,
             root,
-            lookback,
-            db,
+            chain_store,
             chain_config,
             bail: AtomicBool::new(false),
         }
     }
 
+    fn get_lookback_tipset_for_round(&self, height: ChainEpoch) -> anyhow::Result<Cid> {
```

Comment:
This is a misnomer, surely? It's `lookback state root` or something like that?

Reply:
True dat!